lib/tapioca/sorbet_ext/generic_name_patch.rb

```diff
@@ -12,10 +12,11 @@ module Generic
     module TypeStoragePatch
       def [](*types)
         # `T::Generic#[]` just returns `self`, so let's call and store it.
-        constant = super
-        # `register_type` method builds and returns an instantiated clone of the generic type
-        # so, we just return that from this method as well.
-        Tapioca::Runtime::GenericTypeRegistry.register_type(constant, types)
+        super.tap do |constant|
+          # `register_type` method builds and returns an instantiated clone of the generic type
+          # so, we just return that from this method as well.
+          Tapioca::Runtime::GenericTypeRegistry.register_type(constant, types)
+        end
```

Comment:
This comment should probably be updated.

> we just return that from this method as well

We now return `constant`, due to `tap`.

Reply:
This change actually breaks a bunch of other things 😢 ; we're investigating.